source/common/upstream/upstream_impl.h

```diff
@@ -939,6 +926,7 @@ class ClusterInfoImpl : public ClusterInfo,
   const std::unique_ptr<Server::Configuration::CommonFactoryContext> factory_context_;
   std::vector<Network::FilterFactoryCb> filter_factories_;
   Http::FilterChainUtility::FilterFactoriesList http_filter_factories_;
+  bool has_custom_http_filters_{false};
```

Comment:
I don't think it needs to be custom, just configued yeah?
I think it's also worth a comment why we have to latch the bool, e.g. the difference between empty (i.e. implicit codec filter) and present (but maybe not custom configured) codec filter)

Reply:
Maybe this is semantics, but all clusters will have the codec filter configured in http_filter_factories_, but not all clusters will have custom filters specified in the proto config.

Reply:
important semantics!
I still think if we have an HCM filter chain [A B codec] and cluster A with no cluster config and cluster B with cluster config codec filter, requests to cluster A should have A B codec, and requests to cluster B should have codec.
So it's not about if there's more than one filter or a custom filter, it's if there's any actual configuration, vs no configuratoin (which is implicitly codec filter)

Reply:
SGTM. will change.